packages/react-sdk/src/components/component-type.ts

```diff
@@ -1,58 +1,37 @@
 import { z } from "zod";
-import React from "react";
+import type { FunctionComponent } from "react";
 import { IconProps } from "@webstudio-is/icons";
 import type { Style } from "@webstudio-is/css-data";
 
 export type WsComponentMeta<ComponentType> = {
+  /**
+   * container - can accept other components with dnd
+   * control - usually form controls like inputs, without children
+   * embed - images, videos or other embeddable components, without children
+   * rich-text - editable text component
+   * rich-text-child - formatted text fragment, not listed in components list
+   */
+  type: "container" | "control" | "embed" | "rich-text" | "rich-text-child";
+  label: string;
   Component: ComponentType;
-  Icon: React.FunctionComponent<IconProps>;
+  Icon: FunctionComponent<IconProps>;
   defaultStyle?: Style;
-  canAcceptChildren: boolean;
-  // Should children of the component be editable?
-  // Should only be possible for components like paragraph, heading etc.
-  isContentEditable: boolean;
-  // Components that render inside text editor only.
-  isInlineOnly: boolean;
-  // Should be listed in the components list.
-  isListed: boolean;
-  label: string;
   children?: Array<string>;
 };
 
 export const WsComponentMeta = z.lazy(() =>
-  z
-    .object({
-      Component: z.any(),
-      Icon: z.any(),
-      defaultStyle: z.optional(z.any()),
-      canAcceptChildren: z.boolean(),
-      isContentEditable: z.boolean(),
-      isInlineOnly: z.boolean(),
-      isListed: z.boolean(),
-      label: z.string(),
-      children: z.optional(z.array(z.string())),
-    })
-
-    // We need these restrictions because of the limitation of the current drag&drop implementation.
-    // Its position detection logic will be confused if drop target has `string` children.
-    .refine(
-      (val) =>
-        val.isContentEditable === false || val.canAcceptChildren === false,
-      {
-        message:
-          "Content editable componetns are not allowed to accept children via drag&drop, because they may have `string` children",
-        path: ["canAcceptChildren"],
-      }
-    )
-    .refine(
-      (val) =>
-        val.canAcceptChildren === false ||
-        val.children === undefined ||
-        val.children.every((child) => typeof child !== "string"),
-      {
-        message:
-          "Components that can accept children via drag&drop are not allowed to have `string` children",
-        path: ["children"],
-      }
-    )
```

Comment:
Not super important, but maybe we should keep this one

Reply:
We should probably change it to string later and make it work only for rich-text